diff -Nur a/bridge.c b/bridge.c
--- a/bridge.c	2024-10-17 14:31:07.254053339 -0700
+++ b/bridge.c	2024-10-17 14:36:07.641574398 -0700
@@ -35,6 +35,7 @@
 	BRIDGE_ATTR_BRIDGE_EMPTY,
 	BRIDGE_ATTR_MULTICAST_QUERIER,
 	BRIDGE_ATTR_HASH_MAX,
+	BRIDGE_ATTR_HAIRPIN,
 	BRIDGE_ATTR_ROBUSTNESS,
 	BRIDGE_ATTR_QUERY_INTERVAL,
 	BRIDGE_ATTR_QUERY_RESPONSE_INTERVAL,
@@ -58,6 +59,7 @@
 	[BRIDGE_ATTR_BRIDGE_EMPTY] = { "bridge_empty", BLOBMSG_TYPE_BOOL },
 	[BRIDGE_ATTR_MULTICAST_QUERIER] = { "multicast_querier", BLOBMSG_TYPE_BOOL },
 	[BRIDGE_ATTR_HASH_MAX] = { "hash_max", BLOBMSG_TYPE_INT32 },
+	[BRIDGE_ATTR_HAIRPIN] = { "hairpin", BLOBMSG_TYPE_BOOL },
 	[BRIDGE_ATTR_ROBUSTNESS] = { "robustness", BLOBMSG_TYPE_INT32 },
 	[BRIDGE_ATTR_QUERY_INTERVAL] = { "query_interval", BLOBMSG_TYPE_INT32 },
 	[BRIDGE_ATTR_QUERY_RESPONSE_INTERVAL] = { "query_response_interval", BLOBMSG_TYPE_INT32 },
@@ -635,6 +637,10 @@
 	}
 
 	device_set_present(&bst->dev, true);
+	if (bst->config.flags & BRIDGE_OPT_HAIRPIN_MODE) {
+		system_bridge_set_hairpin(bm->dev.dev, bst->config.hairpin);
+	}
+
 	if (!dev->settings.auth || dev->auth_status)
 		device_broadcast_event(&bst->dev, DEV_EVENT_TOPO_CHANGE);
 
@@ -1244,6 +1250,11 @@
 	if ((cur = tb[BRIDGE_ATTR_BRIDGE_EMPTY]))
 		cfg->bridge_empty = blobmsg_get_bool(cur);
 
+	if ((cur = tb[BRIDGE_ATTR_HAIRPIN])) {
+		cfg->hairpin = blobmsg_get_bool(cur);
+		cfg->flags |= BRIDGE_OPT_HAIRPIN_MODE;
+	}
+
 	if ((cur = tb[BRIDGE_ATTR_VLAN_FILTERING]))
 		cfg->vlan_filtering = blobmsg_get_bool(cur);
 }
diff -Nur a/system.h b/system.h
--- a/system.h	2024-10-17 14:31:07.258053386 -0700
+++ b/system.h	2024-10-17 14:31:40.422442046 -0700
@@ -115,6 +115,7 @@
 	BRIDGE_OPT_QUERY_INTERVAL	    = (1 << 2),
 	BRIDGE_OPT_QUERY_RESPONSE_INTERVAL  = (1 << 3),
 	BRIDGE_OPT_LAST_MEMBER_INTERVAL	    = (1 << 4),
+	BRIDGE_OPT_HAIRPIN_MODE		    = (1 << 5),
 };
 
 struct bridge_config {
@@ -138,6 +139,7 @@
 	int hello_time;
 	int max_age;
 	int hash_max;
+	int hairpin;
 
 	bool vlan_filtering;
 };
@@ -247,6 +249,7 @@
 int system_bridge_delbr(struct device *bridge);
 int system_bridge_addif(struct device *bridge, struct device *dev);
 int system_bridge_delif(struct device *bridge, struct device *dev);
+void system_bridge_set_hairpin(struct device *dev, int enable);
 int system_bridge_vlan(const char *iface, uint16_t vid, int16_t vid_end, bool add, unsigned int vflags);
 int system_bridge_vlan_check(struct device *dev, char *ifname);
 void system_bridge_set_stp_state(struct device *dev, bool val);
diff -Nur a/system-linux.c b/system-linux.c
--- a/system-linux.c	2024-10-17 14:31:07.258053386 -0700
+++ b/system-linux.c	2024-10-17 14:31:40.422442046 -0700
@@ -1172,6 +1172,11 @@
 	return 0;
 }
 
+void system_bridge_set_hairpin(struct device *dev, int enable)
+{
+	system_bridge_set_hairpin_mode(dev, enable ? "1" : "0");
+}
+
 int system_if_resolve(struct device *dev)
 {
 	struct ifreq ifr;
