--- a/system-linux.c
+++ b/system-linux.c
@@ -929,6 +929,33 @@ static char *system_get_bridge(const cha
 	return path + 1;
 }
 
+static int system_get_ifindex(const char *name)
+{
+	char path[64], buf[24];
+	int len;
+	FILE *f;
+	struct stat st;
+
+	snprintf(path, sizeof(path), "/sys/class/net/%s/ifindex", name);
+
+	if (stat(path, &st) < 0)
+		return 0;
+
+	f = fopen(path, "r");
+	if (!f) {
+		return 0;
+	}
+
+	len = fread(buf, 1, 23, f);
+	fclose(f);
+
+	if (len <= 0)
+		return 0;
+
+	buf[len] = 0;
+	return strtoul(buf, NULL, 0);
+}
+
 static void
 system_bridge_set_wireless(struct device *bridge, struct device *dev)
 {
@@ -1165,7 +1192,7 @@ int system_if_resolve(struct device *dev
 	struct ifreq ifr;
 
 	if (system_device_ifreq(&ifr, dev->ifname, SIOCGIFINDEX))
-		return 0;
+		return system_get_ifindex(dev->ifname);
 
 	return ifr.ifr_ifindex;
 }
